src/android/BDPointSDKWrapper.java

```diff
@@ -43,666 +47,722 @@
 import android.content.pm.ApplicationInfo;
 import android.graphics.Color;
 import android.os.Build;
-import android.support.v4.app.NotificationCompat;
 
-import android.location.Location;
-import com.google.android.gms.common.GooglePlayServicesNotAvailableException;
-import com.google.android.gms.common.GooglePlayServicesRepairableException;
-import com.google.android.gms.security.ProviderInstaller;
+import androidx.core.app.NotificationCompat;
+import static android.app.Notification.PRIORITY_MAX;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /*
  * @author Bluedot Innovation
  * Copyright (c) 2018 Bluedot Innovation. All rights reserved.
  */
 
-public class BDPointSDKWrapper extends CordovaPlugin implements ServiceStatusListener, ApplicationNotificationListener {
-    public static final String ACTION_AUTHENTICATE = "authenticate";
-    public static final String ACTION_LOGOUT = "logOut";
-    public static final String ACTION_APP_NOTIFICATION_FENCE = "checkedIntoFenceCallback";
-    public static final String ACTION_APP_NOTIFICATION_BEACON = "checkedIntoBeaconCallback";
-    public static final String ACTION_APP_NOTIFICATION_FENCE_CHECKOUT = "checkedOutOfFenceCallback";
-    public static final String ACTION_APP_NOTIFICATION_BEACON_CHECKOUT = "checkedOutOfBeaconCallback";
-    public static final String ACTION_ZONE_INFO_CALLBACK = "zoneInfoCallback";
-    public static final String ACTION_REQUIRE_USER_INTERVENTION_FOR_BLUETOOTH = "startRequiringUserInterventionForBluetoothCallback";
-    public static final String ACTION_REQUIRE_USER_INTERVENTION_FOR_LOCATION = "startRequiringUserInterventionForLocationServicesCallback";
+public class BDPointSDKWrapper extends CordovaPlugin implements InitializationResultListener, ResetResultReceiver {
+    public static final String ACTION_INITIALIZE_SDK = "initializeWithProjectId";
+    public static final String ACTION_RESET_SDK = "reset";
+    public static final String ACTION_IS_INITIALIZED = "isInitialized";
+    public static final String ACTION_ANDROID_START_GEOTRIGGERING = "androidStartGeoTriggering";
```

Comment:
this can be renamed to ACTION_START_GEOTRIGGERING to be in line with other actions of geo-trigger